server/consumer.go

```diff
@@ -2330,8 +2339,10 @@ func (o *consumer) needAck(sseq uint64) bool {
 
 	// Check first if we are filtered, and if so check if this is even applicable to us.
 	if o.isFiltered() && o.mset != nil {
-		var svp StoreMsg
-		sm, err := o.mset.store.LoadMsg(sseq, &svp)
+		svp := storeMsgPool.Get().(*StoreMsg)
+		svp.clear()
```

Comment:
Should this be part of the pool Get() so one does not have to call it every time (and likely forget ;-))?

Reply:
The `Get` method is only called when the pool needs to create a new instance. It doesn’t get called when recycling something that was returned to the pool, so we need to do it this way.

Reply:
You are right!